### ISSUE TITLE
FATAL: Cannot access share. Please check your config.

### ISSUE THREAD

**Issue by tomecki88** [thomasmauerer/hassio-addons#17]
Hi @thomasmauerer,

firstly thanks for an addon like this one.
I'm having problems to connect with my NAS smb share.
In logs I can't find any useful info. Do you know how I can debug to see "WARNING" message (there is nothing in log).
My version of adddon is 2.5.

My HA:
```arch | armv7l
chassis | embedded
dev | false
docker | true
docker_version | 19.03.8
hassio | true
host_os | HassOS 4.11
installation_type | Home Assistant OS
os_name | Linux
os_version | 4.19.127-v7l
python_version | 3.7.7
supervisor | 228
timezone | Europe/Warsaw
version | 0.112.4
virtualenv | false
```
My addon config:
```host: 10.7.0.1
share: Tomecki_NAS
target_dir: /backups/hass
username: hass
password: pass
keep_local: '7'
keep_remote: '30'
trigger_time: '04:00'
trigger_days:
  - Mon
  - Tue
  - Wed
  - Thu
  - Fri
  - Sat
  - Sun
exclude_addons: []
exclude_folders: []
backup_name: '{type} Snapshot (Smb) {date}_HA_{version}'
log_level: debug
```

Log:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[08:30:10] INFO: Host: 10.7.0.1
[08:30:10] INFO: Share: Tomecki_NAS
[08:30:10] INFO: Target Dir: /backups/hass
[08:30:10] INFO: Keep local: 7
[08:30:10] INFO: Keep remote: 30
[08:30:10] INFO: Trigger time: 04:00
[08:30:10] INFO: Trigger days: Mon Tue Wed Thu Fri Sat Sun
[08:30:10] DEBUG: Requested API resource: http://supervisor/services/mqtt
[08:30:10] DEBUG: Request method: GET
[08:30:10] DEBUG: Request data: {}
[08:30:10] DEBUG: API HTTP Response code: 200
[08:30:10] DEBUG: API Response: {"result": "ok", "data": {"host": "core-mosquitto", "port": 1883, "ssl": false, "protocol": "3.1.1", "username": "addons", "password": "veecah4zahfaiBoo4shocha3ienohdoh2fo6leogahz4Moo7noo9Mei7Ziphae1z", "addon": "core_mosquitto"}}
[08:30:11] INFO: Found local mqtt broker - topic is "samba_backup/#"
scripts/helper.sh: line 42: $8: unbound variable
[08:30:11] WARNING: 
[08:30:11] FATAL: Cannot access share. Please check your config.
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```
Thanks in advance for any help.
BR

**Comment by thomasmauerer** [comment]
Hi. Please try without the preceding slash in your target directory:
target_dir: **backups/hass**

Is Tomecki_NAS resolvable on your machine? Maybe you have to configure it via IP address instead.

**Comment by tomecki88** [comment]
Hi, thanks for fast reply.
I changed as you proposed and still I'm getting this same:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[15:54:25] INFO: Host: 10.7.0.1
[15:54:25] INFO: Share: 10.7.0.1
[15:54:25] INFO: Target Dir: backups/hass
[15:54:25] INFO: Keep local: 7
[15:54:25] INFO: Keep remote: 30
[15:54:25] INFO: Trigger time: 04:00
[15:54:25] INFO: Trigger days: Mon Tue Wed Thu Fri Sat Sun
[15:54:25] DEBUG: Requested API resource: http://supervisor/services/mqtt
[15:54:25] DEBUG: Request method: GET
[15:54:25] DEBUG: Request data: {}
[15:54:25] DEBUG: API HTTP Response code: 200
[15:54:25] DEBUG: API Response: {"result": "ok", "data": {"host": "core-mosquitto", "port": 1883, "ssl": false, "protocol": "3.1.1", "username": "addons", "password": "veecah4zahfaiBoo4shocha3ienohdoh2fo6leogahz4Moo7noo9Mei7Ziphae1z", "addon": "core_mosquitto"}}
[15:54:26] INFO: Found local mqtt broker - topic is "samba_backup/#"
scripts/helper.sh: line 42: $8: unbound variable
[15:54:26] WARNING: 
[15:54:26] FATAL: Cannot access share. Please check your config.
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

**Comment by thomasmauerer** [comment]
I guess you read the first version of my comment since you changed the share parameter.

Is Tomecki_NAS the name of the host or the share? I think there is a misunderstanding here: the **host** is the IP address or (resolvable) DNS name of your NAS. The **share** is the name of the share. This parameter must match the name which you configured in your NAS.

**Comment by tomecki88** [comment]
Hi @thomasmauerer,
finally I got some time to debug and I found what is wrong.
I focused on this line in log:
```
[15:54:26] INFO: Found local mqtt broker - topic is "samba_backup/#"
scripts/helper.sh: line 42: $8: unbound variable
```
Made some debug in docker container - manually used smbclient and I can connect with my NAS and finally I found where the problem is.
The problem was that in my password I have "$" sign ;) and to get this working "\" is needed before "$".
Below test:
```
bash-5.0# smbclient -U hass%test$123 \\\\192.168.0.17\\backups
session setup failed: NT_STATUS_LOGON_FAILURE
bash-5.0# smbclient -U hass%test\$123 \\\\192.168.0.17\\backups
Try "help" to get a list of possible commands.
smb: \> 
```
So I added "\" before "$" in my password in configuration and now everything works perfectly.
First backup made and go into NAS drive.

If you will have any free time I think you can add small checking if "$" is in pass and replace with "\$".
Or if any user in future will have similiar problem then should read this issue topic.

Thanks for your addon and great support.
BR

**Comment by thomasmauerer** [comment]
Yes, you're right. I can reproduce it. Thanks a lot for debugging.

The entire password is actually already escaped in quotation marks, but the thing is that bash interprets the dollar sign as a variable even in a string. So it tried to use $8 which is not defined here and so it crashed.

**Comment by thomasmauerer** [comment]
Fixed in v2.6
